yellow-paper/docs/decentralization/actors.md

```diff
@@ -0,0 +1,134 @@
+
+## Node Types
+
+The below node types are ordered in resource requirements. The types exists both in the context of Aztec but also other blockchains. All the nodes, participate in the peer-to-peer (p2p) network, but with varying capacity.
+
+1. **Light Node**: 
+    - Follows the blockchain but only interested in headers (stores headers).
+    - Minimal resource requirements.
+    - Can be used to anchor state membership proofs with hard-to-manipulate state root.
+2. **Pruned Full Node**:
+   - Stores complete state, but only subset of logs and history.
+   - Requires less storage than a Full Node.
+   - Can be used to simulate transactions and view current state.
+2. **Full Node**:
+    - Stores complete state, logs and history.
+    - Requires large storage and a stable internet connection.
+    - Can be used to simulate transactions and view current state and provide history for new nodes to sync.
+3. **Archive Node**:
+    - Stores complete state, logs and history as well as historical state.
+    - Needs significant storage capacity.
+
+When running an Aztec node the actor must connect it to nodes of the underlying blockchain(s). The underlying chains being where data is published and proofs validated. For the following outline, I'm assuming that we just have 1 underlying chain, the Ethereum L1. However, in the future, if data is published on other chains, we will need to extend the constellations with those as well. Also, **blobs** are not stored forever, so what is usually referred to as the history of Ethereum will not longer contain them but just their commitments.
+
+#### Special nodes
+Special nodes are nodes that have specific tasks beyond synchronizing the chain. Namely they participate in block production. In Ethereum they are called validators, but for the sake of briefness, we skip those here, and mainly focus on the Aztec specific nodes.
+
+These nodes are the sequencer and prover nodes. To function they must be connected to a Pruned Full Node, Full Node or Archive Node. 
+
+1. **Sequencer**: 
+   - Requires computational power to simulate blocks.
+   - Needs a robust network to coordinate proof generation and propagate blocks.
+2. **Provers**: 
+   - Have high-performance CPUs/GPUs and a lot of memory.
+   - Assists in proving zero-knowledge (zk) proofs for block validation.
+
+### Aztec Data Retrieval Capabilities:
+An Aztec node needs to be connected to an Ethereum L1 node to learn about the state of the system. Initially synching can be done this way, but due to the validity proofs, it will also support to download the state from a third party and then match it against the L1 state transitioner to validate that the state is correct. 
```

Comment:
This is quite cool. Downloading the full compressed state from something like bittorrent would probably be much faster then asking nodes for individual blobs.

Reply:
Ye. I guess it is not actually because of the validity proofs (unless you make an aggregated like mina that you are having there as well). But more than you can just match the state roots against what is on L1 😎